helper/file/file.go

```diff
@@ -0,0 +1,133 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/samber/lo"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+	ht "github.com/rudderlabs/rudder-server/helper/types"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+
+type fields struct {
+	Input    any         `json:"request"`
+	Output   any         `json:"response"`
+	Metainfo ht.MetaInfo `json:"metadata"`
+}
+
+type fileConfig struct {
+	MaxBatchSize    int
+	MaxBatchTimeout time.Duration
+}
+
+type FileHandle struct {
+	dir                 string
+	fileConfig          *fileConfig
+	logger              logger.Logger
+	fieldsChan          chan fields
+	backgroundCtxCancel context.CancelFunc
+	backgroundWait      func() error
+}
+
+func getFileConfig(parentMod string) *fileConfig {
+	maxBatchSzConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchSize", parentMod)
+	maxBatchTimeoutConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchTimeout", parentMod)
+
+	return &fileConfig{
+		MaxBatchTimeout: config.GetReloadableDurationVar(20, time.Second, maxBatchTimeoutConfPath).Load(),
+		MaxBatchSize:    config.GetReloadableIntVar(200, 1, maxBatchSzConfPath).Load(),
+	}
+}
+
+func WithDirectory(dir string) func(*FileHandle) {
+	return func(fh *FileHandle) {
+		fh.dir = dir
+	}
+}
+
+func WithConfig(maxBatchSz int, maxBatchTimeout time.Duration) func(*FileHandle) {
+	return func(fh *FileHandle) {
+		fh.fileConfig = &fileConfig{
+			MaxBatchSize:    maxBatchSz,
+			MaxBatchTimeout: maxBatchTimeout,
+		}
+	}
+}
+
+func New(parentMod string, opts ...func(*FileHandle)) (*FileHandle, error) {
+	fileConfig := getFileConfig(parentMod)
+	fieldsChan := make(chan fields, fileConfig.MaxBatchSize)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	g, _ := errgroup.WithContext(ctx)
+
+	handle := &FileHandle{
+		dir:                 "mydebuglogs/",
```

Comment:
A few things:
* `parentMod` is used only for configuration, a bit of an overkill to make required parameter
* `path` seems very important in our use case, so making it a required parameter. If path is not configure correctly we might end up filling the disk
* `config.GetReloadableDurationVar` avoid using `Reloadable` variant if it isn't hot reloadable. It can be confusing.
* Always prefer DI config, instead of using package-level functions. Thus I suggest passing conf.Conf

please note tabs are broken:
```suggestion
func WithOptsFromConfig(prefix string, conf config.Config) {
      return func(fh *FileHandle) {
	       maxBatchSizeConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchSize", parentMod)
	       maxBatchTimeoutConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchTimeout", parentMod)


           fh.MaxBatchTimeout = conf.GetDurationVar(20, time.Second, fh.MaxBatchTimeout)
           fh.MaxBatchSize = conf.GetIntVar(200, 1, fh.MaxBatchSize)
    }
}

func New(path string, opts ...func(*FileHandle)) (*FileHandle, error) {
	fileConfig := getFileConfig(parentMod)
	fieldsChan := make(chan fields, fileConfig.MaxBatchSize)

	ctx, cancel := context.WithCancel(context.Background())
	g, _ := errgroup.WithContext(ctx)

	handle := &FileHandle{
		dir:                path,
```